src/components/element-editor/prefill/prefill-child.tsx

```diff
@@ -18,7 +18,7 @@ export const PrefillChild = ({ onDelete, index }: IProps) => {
         formState: { errors },
         control,
         watch,
-    } = useFormContext<IBaseTextBasedFieldValues>();
+    } = useFormContext<TBaseTextBasedFieldValues>();
```

Comment:
I cant recall why we used `TBaseTextBasedFieldValues`, it doesnt represent the actual form values, we should be using `TFormFieldValues` instead

```suggestion
    } = useFormContext<TFormFieldValues>();
```

`TBaseTextBasedFieldValues` and `TBaseOptionGroupBasedFieldValues` should be removed

Reply:
Ok. Changed most of them to `TFormFieldValues`. 

For `validation.tsx` and `valiadation-child.tsx`, I set them as `TOverallTextBasedSchema` since validation is only present for text-based fields.

Additionally, for `conditional-rendering-child.tsx`, if we change it to `TFormFieldValues`, it will cause an error when we are trying to do this


```
setValue(
    `conditionalRendering.${index}.internalId`,
    option.internalId
);
```

since `internalId` is not included in the schema. Should we then add it in since it is already being submitted?

Reply:
ok with both: 
- we are going to refactor the `validation` component so this is fine
- we should add `internalId` indeed, it's good that the type checking catch this